### PR TITLE
[ROCm] add CK GroupNorm to GroupNormTunable

### PR DESCRIPTION
**Pull request by PeixuanZuo**
- Add CK GroupNorm to GroupNormTunable.
- Reduce configuration of GroupNormNHWCOp because CK implementation is better.

The performance gain on stable diffusion v1.5.
Before:
```
'height': 512
'width': 512
'steps': 50
'batch_size': 1
'batch_count': 5
'num_prompts': 1
'average_latency': 2.4782688856124877
'median_latency': 2.4783748388290405
'provider': 'ROCMExecutionProvider'
'disable_safety_checker': True 
```

After:
```
'height': 512, 
'width': 512, 
'steps': 50, 
'batch_size': 1,
'batch_count': 5,
'num_prompts': 1, 
'average_latency': 2.107170510292053,
 'median_latency': 2.1067750453948975,
 'first_run_memory_MB': -1, 
'second_run_memory_MB': -1,
'provider': 'ROCMExecutionProvider', 
'disable_safety_checker': True
```

### CODE REVIEW

File: cmake/external/composable_kernel.cmake [hunk 1]
```diff
@@ -1,5 +1,5 @@
 set(composable_kernel_URL https://github.com/ROCmSoftwarePlatform/composable_kernel.git)
-set(composable_kernel_TAG bef0cb20dba0d9b315df46899310478a81c21852) # 2023-02-16 11:54:08 -0800
+set(composable_kernel_TAG ed3a2e52265e11daa366f47b082141a652b67c58) # 2023-04-10 21:02:17 +0800
```
Comment (tianleiwu) [inline cmake/external/composable_kernel.cmake:2]: Need change cmake/deps.txt and tools/ci_build/github/azure-pipelines/templates/download-deps.yml for updated package?

Comment (PeixuanZuo) [reply cmake/external/composable_kernel.cmake:2]: We don't need to do this for Composable Kernel.

File: onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.cu [hunk 2]
```diff
@@ -37,18 +38,23 @@ Status LaunchGroupNormKernel(
                            "only num_groups=32 is supported. Got", num_groups);
   }
 
-  GroupNormNHWCParams<T> params(nullptr, stream, output, reinterpret_cast<float*>(workspace), input, gamma, beta,
+  GroupNormNHWCParams<T> params(tuning_ctx, stream, output, reinterpret_cast<float*>(workspace), input, gamma, beta,
                                 batch_size, height, width, num_channels, num_groups, epsilon, use_swish_activation);
 
+  if (tuning_ctx != nullptr && tuning_ctx->IsTunableOpEnabled()) {
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.cu:44]: Do we still encounter `tuning_ctx == nulptr`?

File: onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+constexpr size_t kMaxGroupNormBatchSize = 32;
+constexpr size_t kGroupNormNumberOfGroups = 32;
+
+constexpr size_t GetGroupNormWorkspaceSizeInBytes() {
+  // Two buffers for sum and squared sum
+  return (sizeof(float) * 2) * kMaxGroupNormBatchSize * kGroupNormNumberOfGroups;
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.h:24]: Are we using float only for acc type?

File: onnxruntime/contrib_ops/rocm/diffusion/group_norm_ck.cuh [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+#include "contrib_ops/rocm/diffusion/group_norm_common.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+#ifdef USE_COMPOSABLE_KERNEL
+
+template <typename T>
+struct DataTypeAdaptor {
```
Comment (cloudhan) [inline onnxruntime/contrib_ops/rocm/diffusion/group_norm_ck.cuh:26]: Call for refactor in latter PRs, we are duplicating too many of this adaptor 🤦‍♂️.

File: onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.h (hunk 3 above)
Comment (PeixuanZuo) [reply onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.h:24]: Yes, only float for acc type.

File: onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.cu (hunk 2 above)
Comment (PeixuanZuo) [reply onnxruntime/contrib_ops/rocm/diffusion/group_norm_impl.cu:44]: It will not be encountered here, I will remove it.

**Comment by kailums** [comment]
LGTM

**Review by cloudhan — APPROVED** [review]

File: cmake/external/composable_kernel.cmake (hunk 1 above)
Comment (tianleiwu) [reply cmake/external/composable_kernel.cmake:2]: We did that for cutlass. Maybe it is because there is no packaging pipeline for Rocm EP?

**Review by tianleiwu — APPROVED** [review]

File: cmake/external/composable_kernel.cmake (hunk 1 above)
Comment (cloudhan) [reply cmake/external/composable_kernel.cmake:2]: @tianleiwe Maybe we will do it in latter PR. The CK is actively updated, and they don't have a offical stable release at the moment.

**Pull request merged**